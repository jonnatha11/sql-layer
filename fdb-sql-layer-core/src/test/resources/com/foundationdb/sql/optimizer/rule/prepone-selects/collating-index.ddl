CREATE TABLE tc (c1 VARCHAR(10) COLLATE en_US_ci, c2 INT);
CREATE INDEX tc_c1 ON tc(c1);
