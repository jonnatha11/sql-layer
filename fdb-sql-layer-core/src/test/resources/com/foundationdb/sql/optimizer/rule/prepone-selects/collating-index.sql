SELECT c2 FROM tc WHERE c1 > 'A' AND c1 LIKE '%XYZ%'
