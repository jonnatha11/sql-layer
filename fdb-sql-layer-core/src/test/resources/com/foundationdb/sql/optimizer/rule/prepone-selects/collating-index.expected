SelectQuery@420434a7
  ResultSet@221ce1cd[c2]
    Project@162c6b3a[tc.c2]
      Select@3e33b835[]
        Flatten@1ae8fe08(tc)
          Select@367dcd05[like(tc.c1,'%XYZ%')]
            AncestorLookup@7d1f1f35(tc -> [tc])
              SingleIndexScan@63a10c0(Index(test.tc.tc_c1[IndexColumn(c1)]), NONE, >'A', rows = 1, cost = 12.9874)
              TableSource@759715d(tc - TableGroup@4e21b071(tc))